### DISCUSSION TITLE
4.30.1-insiders: The Giscus client is downloaded automatically, although it is excluded from external assets (privacy plugin).

### DISCUSSION THREAD

**Discussion by white-gorilla** [squidfunk/mkdocs-material#5046] [Q&A] [answered]
I have integrated today, with version `4.28.1`, the comment system.
I had previously activated the privacy plugin, without any particular configuration, so that it would automatically download the external assets and set the attributes on the external links.

The giscus widget systematically returned a 404 on loading.
@laymonage told me that it was due to the fact that [I was uploading the script and hosting it on my site](https://github.com/giscus/giscus/issues/900#issuecomment-1427794340).

So I [excluded this dependency as indicated in the documentation](https://github.com/ELS-RD/ai.lefebvre-sarrut/blob/aa82a4c388b3af96c4ed3194b8ee90e633b7f731/mkdocs.yml#L141).
The Giscus client worked with this change.

I upgraded the Insiders version to `4.30.1`, deleted the cache, and now the comment system is returning a 404 again.
When I look at the build log, I see `INFO - Downloading external file: https://giscus.app/client.js` although this is excluded.

Is this a regression?

**Comment by squidfunk** [comment, marked as answer]
Ah, that looks like a regression! I'll look into it.

__Edit__: yes, it was a regression. Fixed in [`0dde44880`](https://github.com/squidfunk/mkdocs-material-insiders/commit/0dde44880b40541c8381b3475c14b69536f7695a) and released as part of [9.0.12+insiders-4.30.2](https://github.com/squidfunk/mkdocs-material-insiders/releases/tag/9.0.12-insiders-4.30.2). If you find other problems, please report them. The privacy plugin was completely rewritten in 4.30 and will now download all assets concurrently – [it's twice as fast on our own documentation, dropping from 28s to 11s on the initial build](https://twitter.com/squidfunk/status/1622606566816980992). It can now also be combined with the [built-in optimize plugin](https://squidfunk.github.io/mkdocs-material/setup/building-an-optimized-site/#built-in-optimize-plugin).
